c10/core/TensorImpl.h

```diff
@@ -455,6 +455,11 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
         key_set_.has(DispatchKey::QuantizedCUDA);
   }
 
+  bool is_xla() const {
+    return key_set_.has(DispatchKey::XLA) ||
+        key_set_.has(DispatchKey::AutogradXLA);
```

Comment:
@ailzhang Thinking aloud: do we really need to test AutogradXLA key? Shouldn't we have an invariant that AutogradXLA key implies XLA key?

Reply:
@ezyang yea nice catch, this line should only check XLA key.

Reply:
Done.